### PR TITLE
HAI-2390 Update nuisance indexes for hanke areas in hanke form areas page

### PR DESCRIPTION
**Pull request by markohaarni**
# Description

Get updated nuisance indexes for hanke area from backend when user has filled all nuisance fields for area, or when area geometry or any nuisance fields are changed.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Other

# Instructions for testing

1. Create new hanke or edit existing
2. Go to areas page of hanke form and draw new hanke area
3. Fill all nuisance fields for new area
4. When all of those are filled, nuisance indexes should be calculated for that area (area color should change and and indexes below fieldset should be updated)
5. Try modifying area geometry or nuisance fields
6. Nuisance indexes should be re-calculated on each change (you can verify this also in browser network tab)

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
      or other location:

# Other relevant info

Please describe here if there is e.g. some requirements for this change or
other info that the tester/user needs to know.

### CODE REVIEW

**Comment by umeetiusbaar** [comment]
Tested and it works. A few notes:
* Selecting an area on the map does not change the area tab - should it?
* Sometimes - not always and cannot reproduce this systematically - when the index color changes while dragging e.g. a corner of an area the dragging halts at that point. Not a big deal.
* Should index 0 be gray instead of green?

**Comment by umeetiusbaar** [comment]
> Tested and it works. A few notes:
> 
> * Selecting an area on the map does not change the area tab - should it?
> * Sometimes - not always and cannot reproduce this systematically - when the index color changes while dragging e.g. a corner of an area the dragging halts at that point. Not a big deal.
> * Should index 0 be gray instead of green?

For the 2nd point, it seems that if altering the area that is also selected as tab, this happens: the dragging of a corner halts when index coloring changes.

File: cypress/integration/E2E.spec.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /// <reference types="cypress" />
+import { HAITTA_INDEX_TYPE, HaittaIndexData } from '../../src/domain/common/haittaIndexes/types';
```
Comment (umeetiusbaar) [inline cypress/integration/E2E.spec.ts:2]: Should we get rid off these Cypress integration tests? They are not used atm, right? And the plan was to use Playwright in E2E testing (when we have time or someone else comes to do it for us)? Are there any benefits of keeping these files in the repo?

File: src/domain/common/haittaIndexes/HaittaIndexNumber.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+type Props = { index?: TormaysIndex; testId?: string };
+
+const HaittaIndexNumber: React.FC<Props> = ({ index, testId }) => {
+  const { t } = useTranslation();
+
+  return (
+    <Box
+      backgroundColor={getColorByStatus(getStatusByIndex(index))}
+      color={getStatusByIndex(index) === LIIKENNEHAITTA_STATUS.YELLOW ? 'black' : 'white'}
```
Comment (umeetiusbaar) [inline src/domain/common/haittaIndexes/HaittaIndexNumber.tsx:19]: Should `getStatusByIndex(index)` be called only once and used the value in these two places?

File: cypress/integration/E2E.spec.ts (hunk 1 above)
Comment (markohaarni) [reply cypress/integration/E2E.spec.ts:2]: Yes, good idea. I was thinking of doing it in its own PR.

Comment (umeetiusbaar) [reply cypress/integration/E2E.spec.ts:2]: Yes, that is a good plan.

File: src/domain/common/haittaIndexes/HaittaIndexNumber.tsx (hunk 2 above)
Comment (markohaarni) [reply src/domain/common/haittaIndexes/HaittaIndexNumber.tsx]: Good point. Fixed that.

**Comment by markohaarni** [comment]
> Tested and it works. A few notes:
> 
> * Selecting an area on the map does not change the area tab - should it?
> * Sometimes - not always and cannot reproduce this systematically - when the index color changes while dragging e.g. a corner of an area the dragging halts at that point. Not a big deal.
> * Should index 0 be gray instead of green?

Fixed these.

**Review by umeetiusbaar — APPROVED** [review]

**Pull request merged**